doc/HttpTransportAdapter.md

```diff
@@ -0,0 +1,192 @@
+
+Each client holds an internal copy of the options passed in the arguments. This means that the options object used to create the client can be disposed. However, since the HTTP transport adapter is a shared pointer, it won't be disposed until all clients using it are disposed first.
+
+## HTTP Transport Adapter Options
+The libcurl and WinHTTP transport adapters can also be initialized with specific options that expose control of features from the HTTP stack underneath it. For example, you can set a proxy or override the default certificate authority (CA) path for libcurl. See the following example:
+```cpp
+  /*
+  * Customize libcurl HTTP transport adapter.
+  */
+
+  // Create the HTTP transport adapter options and set them when initializing the adapter.
+  Azure::Core::Http::CurlTransportOptions curlTransportOptions;
+  curlTransportOptions.Proxy = "https://my.proxy.com";
+  // Create HTTP transport adapter with options.
+  auto curlTransportAdapter = std::make_shared<Azure::Core::Http::CurlTransport>(curlTransportOptions);
+
+  BlobClientOptions options;
+  options.TransportPolicyOptions.Transport = curlTransportAdapter; 
+  auto storageClient = BlobServiceClient(url, credential, options);
+```
+
+The HTTP transport adapter options are specific for each adapter and are optional. This means that if you are creating your own HTTP transport adapter, you don't need to create options for it if you don't want to.
+
+
+## Building a custom HTTP Transport Adapter
+The Azure SDK for C++ uses CMake options to define what HTTP transport adapters to build.
+
+You can see [CMake options](https://github.com/Azure/azure-sdk-for-cpp/blob/master/CONTRIBUTING.md#cmake-build-options) to learn about all the supported options and what is the specific option required to be set when building your own HTTP transport adapter.
+
+Follow next steps to implement your own HTTP Transport Adapter.
+
+1. Implement the interface.
+
+The HTTP transport adapter interface is define in `Azure::Core::HTTP::HttpTransport` from `inc/azure/core/http/transport.hpp`.
+
+```cpp
+/*
+* Implementing the HTTP transport adapter.
+*/
+
+#include "azure/core/http/transport.hpp"
+
+// Derive from HttpTransport
+class CustomTransportAdapter : public Azure::Core::Http::HttpTransport {
+  public:
+    // Override the Send method
+    std::unique_ptr<Azure::Core::Http::RawResponse> Send(
+        Azure::Core::Context const& context,
+        Azure::Core::Http::Request& request) override;
+};
+```
+
+> The implementation needs to take care of using the members from the `Request` argument which contains all the HTTP request information to write the HTTP request using some C++ HTTP stack. Then it needs to create the `RawResponse` from the response and return it. The `context` argument can be optionally used to check if the request is cancelled before performing I/O operations (like writing to the network).
```

Comment:
It's optional because the Transport policy does the check, right?
If so, then it useful only to support cancellation during a synchronous I/O request if the HTTP stack supports this.